src/Symfony/Component/Console/Input/InputOption.php

```diff
@@ -193,6 +200,19 @@ public function getDefault(): string|bool|int|float|array|null
         return $this->default;
     }
 
+    /**
+     * Returns suggested values for input completion.
+     */
+    public function getSuggestedValues(CompletionInput $input): array
+    {
+        $values = $this->suggestedValues;
+        if ($values instanceof \Closure && !\is_array($values = $values($input))) {
+            throw new LogicException(sprintf('Closure for option "%s" must return an array. Got "%s".', $this->name, get_debug_type($values)));
+        }
+
+        return $values;
+    }
```

Comment:
I wonder if the should not replicate the signature of `Commande::complete`.

The feature implemented by this PR is the same, but this allows to pass the `$suggestions` instance to the closure if necessary for future features (like file autocompletion).

```suggestion
    public function complete(CompletionInput $input, CompletionSuggestions $suggestions): void
    {
        $values = $this->suggestedValues;
        if ($values instanceof \Closure && !\is_array($values = $values($input))) {
            throw new LogicException(sprintf('Closure for option "%s" must return an array or null. Got "%s".', $this->name, get_debug_type($values)));
        }

        $suggestions->suggestValues($this->suggestedValues);
    }
```

Reply:
This I don't know. @wouterj your turn for the review :)

Reply:
I implemented this change. I think its necessary to keep this feature open to evolutions.

Reply:
I implemented this change. I think it's necessary to keep this feature open to evolutions.

Reply:
I implemented this change. I think its necessary to keep this feature open to evolutions.

Reply:
I implemented this change. I think it's necessary to keep this feature open to evolutions.